habitat-baselines/habitat_baselines/rl/ppo/single_agent_access_mgr.py

```diff
@@ -0,0 +1,311 @@
+from typing import TYPE_CHECKING, Any, Callable, Dict, Optional
+
+import gym.spaces as spaces
+import numpy as np
+import torch
+from torch.optim.lr_scheduler import LambdaLR
+
+from habitat import logger
+from habitat_baselines.common.baseline_registry import baseline_registry
+from habitat_baselines.common.env_spec import EnvironmentSpec
+from habitat_baselines.common.rollout_storage import (  # noqa: F401.
+    RolloutStorage,
+)
+from habitat_baselines.common.storage import Storage
+from habitat_baselines.rl.ddppo.policy import (  # noqa: F401.
+    PointNavResNetNet,
+    PointNavResNetPolicy,
+)
+from habitat_baselines.rl.hrl.hierarchical_policy import (  # noqa: F401.
+    HierarchicalPolicy,
+)
+from habitat_baselines.rl.ppo.agent_access_mgr import AgentAccessMgr
+from habitat_baselines.rl.ppo.policy import NetPolicy, Policy
+from habitat_baselines.rl.ppo.ppo import PPO
+from habitat_baselines.rl.ppo.updater import Updater
+
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+
+
+@baseline_registry.register_agent_access_mgr
+class SingleAgentAccessMgr(AgentAccessMgr):
+    def __init__(
+        self,
+        config: "DictConfig",
+        env_spec: EnvironmentSpec,
+        is_distrib: bool,
+        device,
+        resume_state: Optional[Dict[str, Any]],
+        num_envs: int,
+        percent_done_fn: Callable[[], float],
+        lr_schedule_fn: Optional[Callable[[float], float]] = None,
+    ):
+        """
+        :param percent_done_fn: Function that will return the percent of the
+            way through training.
+        :param lr_schedule_fn: For a learning rate schedule. ONLY used if
+            specified in the config. Takes as input the current progress in
+            training and returns the learning rate multiplier. The default behavior
+            is to use `linear_lr_schedule`.
+        """
+
+        self._env_spec = env_spec
+        self._config = config
+        self._num_envs = num_envs
+        self._device = device
+        self._ppo_cfg = self._config.habitat_baselines.rl.ppo
+        self._is_distributed = is_distrib
+        self._is_static_encoder = (
+            not config.habitat_baselines.rl.ddppo.train_encoder
+        )
+        self._nbuffers = 2 if self._ppo_cfg.use_double_buffered_sampler else 1
+        self._percent_done_fn = percent_done_fn
+        if lr_schedule_fn is None:
+            lr_schedule_fn = linear_lr_schedule
+        self._init_policy_and_updater(lr_schedule_fn, resume_state)
+
+    def _init_policy_and_updater(self, lr_schedule_fn, resume_state):
```

Comment:
Have you considered making this method a part of `__init__`? 
`_init_policy_and_updater` is called only once (in the `__init__`)

Reply:
I think adding private helper methods like this one helps reading the code, even if this method is only ever called in `__init__`.

Reply:
Yeah I agree with Vince.